### DISCUSSION TITLE
Any way to limit usage of supabase?

### DISCUSSION THREAD

**Discussion by liflovs** [supabase/supabase#1302] [Questions] [answered]
One thing that keeps me from firestore / firebase is that it is impossible to restrict users from abusing the service and hitting the owner's wallet for transactions. So that actually requires building some proxy service and mostly abandoning all "realtiming".
How supabase handles this case? Will "overusage" cause any issue? I see there is no pricing related to this, but this might emerge soon.

**Comment by silentworks** [comment, marked as answer]
On the pricing page, there are some questions similar to this in the FAQ section. There are quite a few answers relating to this and it would seem you wouldn't be charged if you go over your limit but would rather be contacted to discuss moving up to the next tier. Also, you would be notified before hitting the limit and could decide what to do at that point. The database and storage limits are quite high and you aren't charged for API requests. I would assume the Pay as you go would be the next move up after the Pro pricing if you were to go over your limits and the charges per GB seem reasonable in my opinion.

**Reply by liflovs** [reply]
Yep, I guess the answer is that anyone facing this would need to "speak about" the service on "pay as you go". Fair enough.

**Reply by lrnzdc3696** [reply]
Could you link the info on that? I can't find the thing where they say the limit of the Request is.

**Reply by silentworks** [reply]
You can find this on the pricing page under Database, it's the second item. https://supabase.io/pricing
